### PR TITLE
Add check in creating user

### PR DESCRIPTION
**Pull request by MariyaSin**
When you are creating user, you cannot create user with same email, login or phone.

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/673

### CODE REVIEW

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs [hunk 1]
```diff
@@ -44,29 +46,47 @@ public CredentialsResponse Execute(CreateCredentialsRequest request)
...
+                response.Status = OperationResultStatusType.Failed;
+                response.Errors.Add($"Pending user with ID '{request.UserId}' was not found.");
+                return response;
             }
 
-            _userCredentialsRepository.CheckLogin(request.Login, request.UserId);
+            if (_userCredentialsRepository.LoginIsExist(request.Login))
+            {
+                response.Status = OperationResultStatusType.Conflict;
+                response.Errors.Add("The login is busy");
```
Comment (TIove) [inline src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs:62]: mb "The login already exists"?

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs [hunk 2]
```diff
@@ -44,29 +46,47 @@ public CredentialsResponse Execute(CreateCredentialsRequest request)
...
+                return response;
+            }
+
+            if (_userCredentialsRepository.CredentialsIsExist(request.UserId))
```
Comment (TIove) [inline src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs:66]: mb "IsCredentialsExist" or "IsExistCredentials"?

File: src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs [hunk 3]
```diff
@@ -44,29 +46,47 @@ public CredentialsResponse Execute(CreateCredentialsRequest request)
...
+            {
+                response.Status = OperationResultStatusType.Failed;
+                response.Errors.Add("Credentials already exists");
```
Comment (TIove) [inline src/UserService.Business/Commands/Credentials/CreateCredentialsCommand.cs:69]: typo: "exist"

File: src/UserService.Data/Interfaces/IUserCredentialsRepository.cs [hunk 4]
```diff
@@ -26,6 +26,8 @@ public interface IUserCredentialsRepository
         /// <returns>Whether it was successful to edit.</returns>
         bool Edit(DbUserCredentials userCredentials);
 
-        void CheckLogin(string login, Guid userId);
+        bool LoginIsExist(string login);
```
Comment (BelehovEgor) [inline src/UserService.Data/Interfaces/IUserCredentialsRepository.cs:29]: Maybe better is login exist

File: src/UserService.Models.Dto/Enums/OperationResultStatusType.cs [hunk 5]
```diff
@@ -4,6 +4,7 @@ public enum OperationResultStatusType
     {
         FullSuccess,
         PartialSuccess,
-        Failed
```
Comment (BelehovEgor) [inline src/UserService.Models.Dto/Enums/OperationResultStatusType.cs:7]: This enum from kernel, remove here

File: src/UserService.Data.Provider.MsSql.Ef/Migrations/20210708115900_AddUniqueIToLoginProperty.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
+﻿using LT.DigitalOffice.UserService.Models.Db;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace LT.DigitalOffice.UserService.Data.Provider.MsSql.Ef.Migrations
+{
+    [DbContext(typeof(UserServiceDbContext))]
+    [Migration("20210708115900_AddUniqueIToLoginProperty")]
```
Comment (TIove) [inline src/UserService.Data.Provider.MsSql.Ef/Migrations/20210708115900_AddUniqueIToLoginProperty.cs:8]: typo: "AddUniqueId..." probably

File: src/UserService.Data/Interfaces/IUserCredentialsRepository.cs (hunk 4 above)
Comment (TIove) [inline src/UserService.Data/Interfaces/IUserCredentialsRepository.cs:29]: in my opinion IsExistLogin and IsExistCredentials are better

Comment (TIove) [reply src/UserService.Data/Interfaces/IUserCredentialsRepository.cs:29]: In other places u use this naming

**Review by TIove — APPROVED** [review]

File: src/UserService.Models.Dto/Enums/OperationResultStatusType.cs (hunk 5 above)
Comment (MariyaSin) [reply src/UserService.Models.Dto/Enums/OperationResultStatusType.cs:7]: i will

File: src/UserService.Data.Provider.MsSql.Ef/Migrations/20210708115900_AddUniqueIToLoginProperty.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        protected override void Up(MigrationBuilder builder)
+        {
+            builder.AlterColumn<string>(
+                name: nameof(DbUserCredentials.Login),
+                table: DbUserCredentials.TableName,
+                maxLength: 100);
+
+            builder.AddUniqueConstraint(
+                name: $"UX_{nameof(DbUserCredentials.Login)}_unique",
```
[the remark this answers is not in the document]
Comment (MariyaSin) [reply src/UserService.Data.Provider.MsSql.Ef/Migrations/20210708115900_AddUniqueIToLoginProperty.cs]: Done

**Review by BelehovEgor — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

**Pull request merged**